src/util/side_metadata/mod.rs

```diff
@@ -33,64 +114,67 @@
-//!     |        Not Mapped           |
-//!     |                             |
-//!     |_____________________________|     <= offset-l1 = 4MB * Global_worst_case_ratio
-//!     |                             |
-//!     |      PolicySpecific-1       |
-//!     |                             |
-//!     |_____________________________|     <= offset-l2 = offset-l1 + meta_bytes_per_chunk(s3, b3)
-//!     |                             |
-//!     |      PolicySpecific-2       |
-//!     |                             |
-//!     |_____________________________|     <= offset-l3 = offset-l2 + meta_bytes_per_chunk(s4, b4)
-//!     |                             |
-//!     |         Not Mapped          |
-//!     |                             |
-//!     |                             |
-//!     |_____________________________|     <= 4MB * (Global_WCR + PolicySpecific_WCR)
+//! The policy-specific side metadata for 64-bits targets, and the global side metadata for 32-bits targets are used on the same way, except that their base addresses are different.
+//!
+//! Policy-specific side metadata for 32-bits target is slightly different.
+//!
+//! For the first local side metadata bit-set:
+//!
+//! ```
+//! const LOCAL_META_1: SideMetadataSpec = SideMetadataSpec {
+//!    scope: SideMetadataScope::PolicySpecific,
+//!    offset: 0,
+//!    log_num_of_bits: b1,
+//!    log_min_obj_size: s1,
+//! };
+//! ```
+//!
+//! Here, the `offset` is actually the inter-chunk offset of the side metadata from the start of the current side metadata chunk.
+//!
+//! Now, to add a second side metadata bit-set, offset needs to be calculated based-on the first global bit-set:
+//!
+//! ```
+//! const LOCAL_META_2: SideMetadataSpec = SideMetadataSpec {
+//!    scope: SideMetadataScope::PolicySpecific,
+//!    offset: LOCAL_META_1.offset + meta_bytes_per_chunk(LOCAL_META_1)
+//!    log_num_of_bits: b2,
+//!    log_min_obj_size: s2,
+//! };
+//! ```
 //!
 //! So far, no metadata space is allocated.
 //!
-//! For this purpose, each plan should override `fn global_side_metadata_per_chunk(&self) -> usize;` to return the size of the global side metadata it needs per chunk. This can be calculated using the `meta_bytes_per_chunk` function.
+//! For this purpose, each plan should override `fn global_side_metadata_spec_vec(&self) -> &'static Vec<SideMetadataSpec>` to return a vector of the global side metadata it needs.
 //!
-//! For the local metadata bit-sets, each policy needs to follow the same pattern as the global metadata, with two differences:
+//! This function can be implemented like:
+//! ```
+//! return Arc::new(vec![GLOBAL_META_1, GLOBAL_META_2]);
+//! ```
 //!
-//! 1. scope should be `SideMetadataScope::PolicySpecific`,
-//! 2. each policy needs to override `fn local_side_metadata_per_chunk(&self) -> usize;
+//! For the local metadata bit-sets, each policy needs override `fn global_side_metadata_spec_vec(&self) -> &'static Vec<SideMetadataSpec>`.
```

Comment:
I think this should be `fn local_side_metadata_spec_vec(&self)`

Reply:
Right,
I will push many changes soon and this will also be fixed